### DISCUSSION TITLE
--total_silence feature

### DISCUSSION THREAD

**Discussion by MilkAndBanana1** [WyattBlue/auto-editor#608] [2. Feature Ideas]
I had this idea the other day which I think would be sick. I find often times that when I run it through this tool that I often only want it to cut after a certain amount of collective silence and not necessarily if it just reaches a certain amount of time per instance of silence. A good use case for this is for more conservative editing styles where pauses and space is given with their style of commentary.

```shell
auto-editor video.mp4 --edit audio:0.03,total_silence=10sec
``` 
For example, in this command the auto-editor will only cut out an instance of silence when it reaches the threshold after it has collected 10 seconds of silence before that instance. It would then also trigger immediately if the silence is longer than the total_silence and would reset the accordingly.

If there is already a way to do this through cmd or if there's a better name for this because this isn't the best word choice due to it being confusing that would be great.

**Comment by WyattBlue** [comment]
I don't understand how you would actually implement this.

`--edit` is a LISP language. I don't even need to add anything. You can just implement it yourself.

```
auto-editor video.mp4 --edit "(total-silence audio:0.03 10sec)"
```

```
(define (total-silence arr time)
  ; I don't know how you plan to mutate `arr`.
  arr
)

**Reply by MilkAndBanana1** [reply]
Revisiting this. If it is already possible currently, my guess would be the following:

arr = time of silent and non-silent in frames, [silent frames , non-silent frames]
total-silence = defaults to amount of frames, for practicality convert seconds to frames
goes through array and compares each consecutive silent and non-silent instance if it reaches silent or total-silence threshold 

ex:
```
total-silence = 1000 frames
silent threshold = 2000 frames
arr = [200,10000] , [500,15000] , [1000,8000] , [4000,17000] , [500,12000] , [600,18000]
                                     ^total-s     ^silent-t                    ^total-s
```
I have no idea how arr is structured but this is in my head how it would be changed after analysing the audio, and then if total-silence is enabled, it will go through each instance and run through which silent instances can be cut through total-silence's threshold.

**Reply by MilkAndBanana1** [reply]
Wait I think I just realized, is minclip this exact feature?

So like mincut is the minimum amount of silent it requires (silent threshold) and then minclip is the required amount of non-silent before a cut (total silence)?
